src/Application/Documents/Queries/GetDocumentsByTitle/GetDocumentsByTitleQuery.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Application.Common.Interfaces;
+using Application.Common.Mappings;
+using Application.Common.Models;
+using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Documents.Queries.GetDocumentsByTitle;
+
+public record GetDocumentsByTitleQuery : IRequest<PaginatedList<DocumentDto>>
+{
+    public Guid UserId { get; init; }
+    public string? SearchTerm { get; init; }
+    public int Page { get; init; } = 1;
+    public int Size { get; init; } = 10;
+}
+
+public class GetDocumentsByTitleQueryHandler : IRequestHandler<GetDocumentsByTitleQuery, PaginatedList<DocumentDto>>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public GetDocumentsByTitleQueryHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<PaginatedList<DocumentDto>> Handle(GetDocumentsByTitleQuery request, CancellationToken cancellationToken)
+    {
+        var currentUser = await _context.Users
```

Comment:
You are checking current user through `UserId` in Request message which is an untrusted data from user.

Reply:
This is violating `Confidentiality` in CIA triad